002_simulation_structure/002_communication.tex

```diff
@@ -0,0 +1,205 @@
+\documentclass{article}
+\usepackage{longtable}
+\usepackage[a4paper, top=3cm, bottom=3cm, left=2.5cm, right=2.5cm, marginparwidth=2cm, headsep=1.2cm]{geometry} % layout
+
+\begin{document}
+
+\section{Primitive Messaging}
+An important role of Team 1 was to design a mechanism for agents to talk amongst each other in the simulation that would allow automated communication. The first step in this process was to design a system which would allow an agent to pass along a minimum viable message to their neighbours. These were the goals of communication in the MVP:
+\begin{itemize}
+    \item Agents could only talk to their immediate neighbors (+/- 1 floor)
+    \item Agents should be given the ability to ignore communication
+    \item Multiple agents could be “speaking” at a given tick
+    \item One agent could send one message per tick
+    \item \textbf{It is entirely an agent's discretion if/when/how they wish to react to a message as long as their behavior is not deceptive}
+\end{itemize}
+As a team, Team 1 emphasized the last point. The infrastructure should not force any behavior onto the agent as long as the behavior was honest. The basic philosophy of the communication infrastructure was to dictate as little as possible about the agent's behavior while keeping the API friendly, readable and hard to get wrong. This usually meant making providing basic override-able behaviors to be as unobtrusive as possible to team strategies. \newline
+The minimum default message had to have the following fields that are accessible to all the agents that receive it:
+\begin{itemize}
+    \item \textbf{\texttt{MessageType}}: Indicates one of 13 enumerated message types.
+    \item \textbf{\texttt{SenderFloor}}: Returns the floor this message was sent from.
+    \item \textbf{\texttt{TargetFloor}}: The floor the message is addressed to.
+    \item \textbf{\texttt{ID}}: Each message sent is given a unique ID.
```

Comment:
might need to change `\texttt{}` to `\textit{}`, depending on the final choice of the editors